AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSPluginsCore
+@testable import AWSDataStoreCategoryPlugin
+
+class SyncEventEmitterTests: XCTestCase {
+    var initialSyncOrchestrator: MockAWSInitialSyncOrchestrator?
+    var reconciliationQueue: MockAWSIncomingEventReconciliationQueue?
+    var syncEventEmitter: SyncEventEmitter?
+    var expectedModelSyncedEventPayloads: [ModelSyncedEvent]!
+
+    override func setUp() {
+        super.setUp()
+
+        ModelRegistry.reset()
+        MockModelReconciliationQueue.reset()
+        MockAWSInitialSyncOrchestrator.reset()
+
+        expectedModelSyncedEventPayloads = [ModelSyncedEvent(modelName: "Post",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0),
+                                            ModelSyncedEvent(modelName: "Comment",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0)]
+    }
+
+    /// - Given: A SyncEventEmitter
+    /// - When:
+    ///    - One model is registered
+    ///    - Perform an initial sync
```

Comment:
more about this test cares about
```suggestion
    ///    - Initial sync occured
    ///    - model reconcillation occured
```

Reply:
updated